shimming-toolbox/shimmingtoolbox/cli/b0shim.py

```diff
@@ -867,84 +881,70 @@ def realtime_dynamic(fname_fmap, fname_anat, fname_mask_anat_static, fname_mask_
 
         # If it's a scanner
         if type(coil) == ScannerCoil:
-            if coil in list_coils_common:
-                keys = [str(order) for order in AVAILABLE_ORDERS
-                        if (order != -1 and (str(order) in coil_indexes_riro[coil.name]
-                                             or str(order) in coil_indexes_static[coil.name]))]
-            elif coil in coil_static_only:
-                keys = [str(order) for order in AVAILABLE_ORDERS
-                        if (order != -1 and str(order) in coil_indexes_static[coil.name])]
-            elif coil in coil_riro_only:
-                keys = [str(order) for order in AVAILABLE_ORDERS
-                        if (order != -1 and str(order) in coil_indexes_riro[coil.name])]
-
-            save_coefs_static = None
-            save_coefs_riro = None
-            for key in keys:
-                if coil in list_coils_riro:
-                    if key in coil_indexes_riro[coil.name]:
-                        coefs_coil_riro = copy.deepcopy(
-                            coefs_riro[:, coil_indexes_riro[coil.name][key][0]:coil_indexes_riro[coil.name][key][1]])
+            if 'hrd' in o_format_sph:
+                logger.debug("Converting Siemens scanner coil from Shim CS (LAI) to Gradient CS")
+                
+                def _coefs_to_dict(coefs_coil, scanner_coil_order, manufacturer):
+                    # Fill in a dictionary with the coefficients for each order
+                    coefs_scanner = {}
+                    start_channel_scanner = 0
+                    for order in scanner_coil_order:
+                        end_channel_scanner_order = start_channel_scanner + channels_per_order(order, manufacturer)
+                        coefs_scanner[order] = coefs_coil[:, start_channel_scanner:end_channel_scanner_order]
+                        start_channel_scanner = end_channel_scanner_order
+                    coefs_coil = coefs_scanner
+                    
+                    return coefs_coil
```

Comment:
This bit of code seems to also be used for non RT shimming, maybe move it to the global scope and reuse it in the "normal" shim procedure.

Reply:
see [9fa9b62](https://github.com/shimming-toolbox/shimming-toolbox/pull/586/commits/9fa9b62774a6ed9f30e6144aa29c2d74d0c6d371)